kuiper-game/Domain/GameTime.cs

```diff
@@ -1,35 +1,43 @@
 using System;
+using Microsoft.VisualBasic;
 
-public struct GameTime
+namespace Kuiper.Domain
 {
-    private static readonly DateTime gameStartDate = new DateTime(2078, 1, 1);
-    private const long tickAccelerationConstant = 7; //1 real day, is 7 game days
-    public static DateTime RealStartTime {get; set;}
+    public readonly struct GameTime
+    {
+        private static readonly long GameTimeEpoch = new DateTime(2078, 1, 8).Ticks;
+        private const long TickAccelerationConstant = 7; //1 real day is 7 game days
+        public static long RealTimeEpoch { get; private set; }
+        public long Ticks { get; }
 
-    public static DateTime GameStartDate {
-        get
+        /// <summary>
+        /// Initialize a game time from a real time
+        /// </summary>
+        /// <param name="realTimeEpoch">The starting timestamp in real time UTC. This is equal to gametime GT 2078-01-01</param>
+        /// <param name="ticks">Additional ticks</param>
+        public GameTime(long realTimeEpoch, long ticks = 0)
         {
-            return gameStartDate;
+            RealTimeEpoch = realTimeEpoch;
+            Ticks = ticks;
         }
-    }
 
-    public static TimeSpan ElapsedGameTime {
-        get 
+        public GameTime Add(long ticks)
         {
-            var elapsedRealTime = DateTime.Now.Subtract(RealStartTime);
-            var elapsedGameTime = new TimeSpan(elapsedRealTime.Ticks * tickAccelerationConstant);
-
-            return elapsedGameTime;
+            return new(RealTimeEpoch, ticks);
```

Comment:
I really dont understand why the dotnet team added the `new()`, supposedly for when the type is obvious (like the original intent behind var) but at this point, what type is it? (its `GameTime`) but its hard to read (for me atleast)

Reply:
heh - I agree. This was actually Rider suggesting it to me, and I was kinda like - "yeah, seems legit" 😉